bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/ManagedRuleProvider.java

```diff
@@ -24,16 +24,16 @@
  */
 public class ManagedRuleProvider extends AbstractManagedProvider<Rule, String, RuleDTO>implements RuleProvider {
 
-    private RuleManager ruleManager;
+    private RuleEngine ruleEngine;
```

Comment:
There should be no dependency between the ManagedRuleProvider and the RuleEngine. The ManagedRuleProvider should just be treated like any other rule provider. The RuleRegistryImpl should instead overwrite the add, update, remove methods from AbstractRegistry and sync this to the RuleEngine accordingly.